v4/examples/langchain_langsmith.mdx

```diff
@@ -0,0 +1,428 @@
+  - Chat History is where we will inject all previous messages so that the agent has context on what was said previously.
+  - Input is new input from the end user.
+- We then instantiate our GPT3.5 model that will be the LLM we will be using. You can swap this our with Antrophic or any other provider just by replacing this one line - LangChain makes this seamless.
+- Lastly, we join this all together with our tools to create an agent executor.
+
+### Setup Chatbot
+
+The above code is static in that it will only reply to our first question but we might need to have a conversation in order to find a time that suits both the user and my schedule. We therefore need to create a chatbot with tool calling capabilities and the ability to remember past messages. LangChain supports this with RunnableWithMessageHistory().
+
+It essentially allows us to store the previous replies of our conversation in a chat_history variable (mentioned above in our prompt template) and tie this all to a session identifier so your API can remember information pertaining to a specific user/session. Below is our code in order to implement this:
+
+```python
+from langchain.memory import ChatMessageHistory
+from langchain_core.runnables.history import RunnableWithMessageHistory
+
+demo_ephemeral_chat_history_for_chain = ChatMessageHistory()
+conversational_agent_executor = RunnableWithMessageHistory(
+    agent_executor,
+    lambda session_id: demo_ephemeral_chat_history_for_chain,
+    input_messages_key="input",
+    output_messages_key="output",
+    history_messages_key="chat_history",
+)
+```
+
+Let us run a simple local test to make sure everything is working as expected.
+
+```python
+class Item(BaseModel):
+    prompt: str
+    session_id: str
+
+def predict(item, run_id, logger):
+    item = Item(**item)
+
+    output = conversational_agent_executor.invoke(
+        {
+            "input": user_input,
+        },
+        {"configurable": {"session_id": item.session_id}},
+    )
+
+    return {"result": output} # return your results
+
+if __name__ == "__main__":
+    while True:
+        user_input = input("Enter the input (or type 'exit' to stop): ")
+        if user_input.lower() == 'exit':
+            break
+        result = predict({"prompt": user_input, "session_id": "12345"}, "test", logger=None)
+        print(result)
+```
+
+The above code does the following:
+
+- We define a Pydantic object which specifies the parameters our API expects - the user prompt and a session id to tie the conversation to.
+- The predict function in Cerebrium is the entry point for our API so we just pass the prompt and session id to our agent and print the results.
+  ‍
+
+To run this, simply run the following in your terminal: python main.py. You will need to replace your secrets with the actual values when running locally. You should then see output similar to the following:
```

Comment:
May want to also include that users should manually install pip packages added to toml if they don't have those dependencies. PS: You also need langchain-community to run this. See error below 
```ModuleNotFoundError: Module langchain_community.chat_message_histories not found. Please install langchain-community to access this module. You can install it using `pip install -U langchain-community` ```

```suggestion
To run this, simply install the pip dependencies manually by typing the following into your terminal `pip install pydantic langchain pytz openai langchain_openai langchain-community` and then run `python main.py` to execute your main python file. You will need to replace your secrets with the actual values when running locally. You should then see output similar to the following:
```